File: c++/advanced/3b_longest_substring_without_repeating_characters.cpp
//Find Length of Longest Substring Without Repeating Characters

int lengthOfLongestSubstring(string s) {
        unordered_set<char> st;
        int ans = 0, l=0;
        for(int r=0; r<s.size(); r++){
            char x = s[r];
            if(st.find(x) == st.end()){
                st.insert(x);
            }
            else{
                while(s[l] != x){
                    st.erase(s[l]);
                    l++;
                }
                l++;
            }
            ans = max(ans, r-l+1);
        }
        return ans;
}

File: js/advanced/3b_longest_substring_without_repeating_characters.js
//Find Length of Longest Substring Without Repeating Characters

var lengthOfLongestSubstring = function(s) {
    let l=0, ans=0
    const st = new Set() 
    for(let r=0; r<s.length; r++){
        let x = s[r]
        if(!st.has(x)){
            st.add(x)
        }
        else{
            while(s[l] != x){
                st.delete(s[l])
                l++
            }
            l++
        }
        ans = Math.max(ans, r-l+1)
    }
    return ans
};